utils/bin/indexer/src/main.rs

```diff
@@ -0,0 +1,100 @@
+use std::{cmp::min, collections::HashMap};
+
+use alloy::{
+    primitives::{Address, Bytes},
+    providers::{Provider, ProviderBuilder},
+    rpc::types::Filter,
+    sol,
+    sol_types::{SolEvent, SolValue},
+};
+
+// Codegen from ABI file to interact with the contract.
+sol!(
+    #[allow(missing_docs)]
+    #[allow(clippy::too_many_arguments)]
+    #[sol(rpc)]
+    ComposableCoW,
+    "../../abi/ComposableCoW.json"
+);
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    const WINDOW_SIZE: u64 = 10000;
+    const MAINNET_DEPLOYMENT_GENESIS: u64 = 19277205;
+    const MAINNET_CONSTANT_PRODUCT_HANDLER: &str = "0x34323B933096534e43958F6c7Bf44F2Bb59424DA";
+    const GNOSIS_DEPLOYMENT_GENESIS: u64 = 32478660;
+    const GNOSIS_CONSTANT_PRODUCT_HANDLER: &str = "0xB148F40fff05b5CE6B22752cf8E454B556f7a851";
+
+    let rpc_url = match std::env::var("RPC_URL") {
```

Comment:
IMO it's fine to just `.unwrap()` on initialization code. This would also result in an abort and a good error message.

Alternatively since you already return an error from `main` you could also use `?` here (might require to use `.wrap_err("some error message")` first; not familiar with `eyre`).

Also what was the reason to go with `eyre`? IMO it's fine to try out different things in these little helper tools. Just curious if there was a need that `anyhow` was not suitable for.

Reply:
No specific reason for eyre over anyhow.